src/diagnostics/wizardSteps/PromptSlotSelection.ts

```diff
@@ -49,22 +51,24 @@ export class PromptSlotSelection extends WizardStep {
             }
             throw e;
         }
-
-        this._wizard.selectedDeploymentSlot = pickedItem.data;
+        this._wizard.selectedDeploymentSlotTreeItem = pickedItem.data;
         this._wizard.writeline(`The deployment slot you selected is: ${this._wizard.selectedDeploymentSlot.name}`);
     }
+
     /**
      * Returns all the deployment slots and the production slot.
      */
-    private async getDeploymentSlots(): Promise<Site[]> {
-        const client = this._wizard.websiteManagementClient;
-        const allDeploymentSlots = await callWithTimeout(
-            () => {
-                return client.webApps.listByResourceGroup(this.site.resourceGroup, { includeSlots: true });
-            },
-            DEFAULT_TIMEOUT);
-        return allDeploymentSlots.filter((slot) => {
-            return slot.repositorySiteName === this.site.name;
-        });
+    private async getDeploymentSlotsTreeItems(): Promise<SiteTreeItem[]> {
+        const appServiceTreeItem = this._wizard.uiTreeItem.treeItem;
+        const result = await appServiceTreeItem.loadMoreChildren(<IAzureParentNode<SiteTreeItem>>this._wizard.uiTreeItem);
```

Comment:
You should never call `treeItem.loadMoreChildren` directly. Instead change the type of `this._wizard.uiTreeItem` to `IAzureParentNode<SiteTreeItem>` and call `this._wizard.uiTreeItem.getCachedChildren()`

If you call 'loadMoreChildren', then you only get the "next batch" to be displayed in the tree. However, if you call 'getCachedChildren', you get all of the nodes that have been loaded so far. The operation is potentially much faster if the user has already opened the slots in the tree

Reply:
NOTE: This isn't really an issue in the case of slots because it always load all of the slots at once (aka, slots doesn't support a continuation token or load more of any sort). It's just the principle of the thing haha

Reply:
I see. Will change to `getCachedChildren()`. 

To your other FYI for AzureTreeDataProvider.showNodePicker. I really tried that and felt it useful, but unfortunately it does not give a desired behavior, in that, whatever context value I use, it always prompts user to select something starting from selecting an subscription, then a App Service, then a deployment slot. This won't work for us, because user asks to initiate a command from an already selected AppService node, we would like her to select a underneath deployment slot directly. 

Also our selection logic requires the list to include the app service itself. From my observation, when it gets to the deployment slots level, it only displays the real deployment slots, and will leave the parent app service out.

Reply:
I will keep your scenario in mind for 'showNodePicker' and let you know if/when we add support for it. If you want that behavior, then it's probably useful to other people as well

Reply:
Thanks @EricJizbaMSFT Much appreciated

Reply:
I don't think we need to worry about a large number of children in deployment slots. I think the slot limit on standard websites is 4.